### ISSUE TITLE
EBS-Driver fails to mount AWS volume

### ISSUE THREAD

**Issue by dadarakt** [rancher/rancher#12106]
**Rancher versions:**
rancher/server: 1.6.14
rancher/agent: 1.2.9

**Infrastructure Stack versions:**
healthcheck: 0.3.3
ipsec: 0.13.7
network-services: 0.9.5
scheduler: 0.8.3
ebs-driver: 0.8.11

**Docker version**:  Server Version: 17.09.1-ce

**Operating system and kernel:** 4.9.77-31.58.amzn1.x86_64

**Type/provider of hosts:** AWS, hvm-hosts

**Setup details:** single node rancher, internal DB

**Environment Template:** Cattle 0.183.37

**Steps to Reproduce:**
After setting up the EBS driver for a stack, it is possible to create volumes for stacks in docker-compose and use it, i.e.:
```yml
volumes:
  test-data:
    volumeType: gp2
    size: 10
```
Snapshots created from this image could also be used to create new volumes like so:
```yml
volumes:
  test-data:
    ec2_az: <availability-zone>
    snapshotID: <snapshot-ID>
    size: 10
```
When creating a snapshot off an existing EC2 volume used outside of Rancher or when creating a new volume in the AWS EC2 console in the right availability zone with the exact same parameters as above and then trying to use it in Rancher leads to the problem.
```yml
volumes:
  test-data:
    ec2_az: <availability-zone>
    volumeID: <volume-ID>
    size: 10
```

**Results:**
The EBS driver on the Rancher hosts fails to mount the drive, reporting:
```
mount: wrong fs type, bad option, bad superblock on /dev/xvdg,
       missing codepage or helper program, or other error

       In some cases useful info is found in syslog - try
       dmesg | tail or so.
```
When manually connecting to a host and attaching the AWS-created volume to `/dev/xvdg` and attaching the rancher-created volume to `/dev/xvdf`  using the aws-cli, I can reproduce that behavior:
```
$ sudo mount /dev/xvdg <mount-point>
mount: wrong fs type, bad option, bad superblock on /dev/xvdg,
       missing codepage or helper program, or other error

       In some cases useful info is found in syslog - try
       dmesg | tail or so.
```
Investigating leads to the following:
```
$ lsblk
NAME    MAJ:MIN RM SIZE RO TYPE MOUNTPOINT
xvda    202:0    0   8G  0 disk 
└─xvda1 202:1    0   8G  0 part /
xvdg    202:96   0  20G  0 disk 
└─xvdg1 202:97   0   8G  0 part 
xvdf    202:80   0   1G  0 disk /root/rancher_created_volume
```
So the problem is, that AWS creates a partition on the volume and rancher does not.

Shouldn't the EBS driver support volumes that are created using the de-facto AWS standard?
Or is there a reason not to mount the primary partition?

It would really help me in production use-cases, where I don't need to re-create existing volumes and manually copy data there. 

I'm also happy for any workarounds, thanks!

**Comment by ahaynssen** [comment]
@dadarakt wasn't sure if you saw my reference, but you can pull the branch above and build yourself to solve this issue.

**Comment by soumyalj** [comment]
Tested with ebs version rancher/storage-ebs:v0.9.7.
1. Created a volume in EBS and attached the volume to an AWS instance
2. Partitioned the volume and performed mkfs

Used the volume id in creating another stack in rancher and the stack was created successfully.
So, if the volume is partitioned it can be used in rancher ebs.
Also created a snapshot from the volume created above and used the snapshot  id  in a stack successfully.
```
version: '2'
services:
  foo:
    image: ubuntu 
    stdin_open: true
    volumes:
    - bar:/test
volumes:
  bar:
    driver: rancher-ebs
    driver_opts:
      size: '10'
      snapshotID: "snaptest"

```
In v1.6.14, with the same steps, the containers were stuck in activating state. 

@dadarakt:
Can you explain how did u create a volume outside of rancher and a snapshot out of it? 
Can you check if u see the same issue if an mkfs is done after u create a new volume in AWS and then use the volume in your tests.

@StrongMonkey : Can you confirm if rancher supports using a volume directly without performing "mkfs" on the volume ?

**Comment by StrongMonkey** [comment]
@dadarakt we have merged @ahaynssen 's fix into rancher/storage-ebs:v0.9.7 so that you should be able to use an existing EBS volume with partition. If you are using an existing volume, make sure it is formatted as rancher-ebs-driver won't format it for you.

**Comment by deniseschannon** [comment]
@dadarakt if you are still having issues, please re-open.
